frontend/src/container/MetricsExplorer/Explorer/utils.tsx

```diff
@@ -35,3 +55,43 @@ export const splitQueryIntoOneChartPerQuery = (query: Query): Query[] => {
 
 	return queries;
 };
+
+/**
+ * Hook to get data for multiple metrics with a synchronous loading and error state
+ * @param metricNames - The names of the metrics to get
+ * @param isEnabled - Whether the hook is enabled
+ * @returns The loading state, the metrics data, and the error state
+ */
+export function useGetMetrics(
+	metricNames: string[],
+	isEnabled = true,
+): {
+	isLoading: boolean;
+	isError: boolean;
+	metrics: (MetricMetadata | undefined)[];
+} {
+	const metricsData = useGetMultipleMetrics(metricNames, {
```

Comment:
Isn't there a BE api just for this? We should create one if so.
We should avoid multiple calls wherever we can. This is one of those cases it feels like

Reply:
We don't have an API for fetching metadata of multiple metrics at once. Just one at a time. The API in this case should not be that heavy, so it should be fine I think.
@srikanthccv may be able to answer this better.